sycl/include/CL/sycl/ONEAPI/intel_matrix/matrix-amx.hpp

```diff
@@ -0,0 +1,439 @@
+//==------------- submatrix-amx.hpp - SYCL matrix -----------*- C++ -*---==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// ===--------------------------------------------------------------------=== //
+///
+/// We provide new interfaces for matrix muliply in this patch:
+/// 1. A new class called joint_matrix is introduced, and The user needs to
+/// specify the type of the elements, sizes, and the memory layout.
+///
+/// 2. joint_matrix_load is used for loading data from main momory to tiles of
+/// AMX or kernel's local memory.
+///
+/// 3. joint_matrix_store is used for storing data tiles of AMX or kernel's
+/// local memory to main memory.
+///
+/// 4. joint_matrix_mad is used for the matrix multiply and add function.
+/// It performs the multiply operation on the matrices A and B, accumulates the
+/// result with C and return the result.
+///
+/// The following operation can be realized with the interfaces:
+///  C = A*B+C
+/// 1. All cases where A(int8, any-size, row_major), B(int8, any-size,
+/// packed_b), C(int32, any-size, row_major)
+/// 2. All cases where A(bf16, any-size, row_major), B(bf16, any-size,
+/// packed_b), C(float, any-size, row_major)
+///
+///
+// ===--------------------------------------------------------------------=== //
+
+#pragma once
+
+#include <immintrin.h>
+
+__SYCL_INLINE_NAMESPACE(cl) {
+namespace sycl {
+namespace ext {
+namespace intel {
+namespace detail {
+template <typename T> class submatrix {
+public:
+  _tile1024i tile;
+  short rows, cols;
+};
+
+constexpr size_t dynamic_extent = std::numeric_limits<size_t>::max();
+
+template <typename T> struct elems_per_dword {
+  static constexpr size_t value = 1;
+};
+
+#define ELEMS_PER_DWORD(TYPE, NUM)                                             \
+  template <> struct elems_per_dword<TYPE> {                                   \
+    static constexpr size_t value = NUM;                                       \
+  };
+
+ELEMS_PER_DWORD(int8_t, 4)
+ELEMS_PER_DWORD(__bfloat16, 2)
+
+} // namespace detail
+} // namespace intel
+} // namespace ext
+} // namespace sycl
+
+namespace sycl {
+namespace ext::intel {
```

Comment:
Here we close `sycl::ext::intel` namespace just to open it again.
Should we just remove this code?
```suggestion
```